### PR TITLE
Thing support in rules

### PR DESCRIPTION
**Pull request by seime**
See updated README.md

* Generates JRuleThing/JRuleThings files that may be referenced in rules.
* Support for enabling/disabling/restarting things.
* Support for fetching bridge of bridged items + bridged items from bridge

Trigger support to be improved, JRuleEvent needs some refactoring/splitting.

Solves #41

### CODE REVIEW

File: src/main/java/org/openhab/automation/jrule/things/JRuleBridgeThing.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jrule.things;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The {@link JRuleBridgeThing} thing represents a thing that is connected to a bridge
```
Comment (querdenker2k) [inline src/main/java/org/openhab/automation/jrule/things/JRuleBridgeThing.java:19]: JRuleBridgeThing and JRuleStandaloneThing do per JavaDoc the same?

File: src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java [hunk 2]
```diff
@@ -237,13 +276,14 @@ private boolean initializeFolder(String folder) {
...
+        thingGenerator.generateThingsSource(thingRegistry.getAll());
         Boolean result = false;
-        if (compiler.compileItems()) {
+        if (compiler.compileItemsAndThings()) {
 
-            logInfo("Creating items jar");
+            logInfo("Creating items and things jar");
             JRuleUtil.createJarFile(config.getItemsRootDirectory(),
                     compiler.getJarPath(JRuleCompiler.JAR_JRULE_ITEMS_NAME));
             result = true;
```
Comment (querdenker2k) [inline src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java:289]: Wouldn't it be better to either generate a separate JAR file or rename the current one?

**Review by querdenker2k — APPROVED** [review]

File: src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java (hunk 2 above)
Comment (seime) [reply src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java:289]: Yes, I think we should rename it, and not create separate files (with compiled actions we'll have 3 or more jar files to joggle)

File: src/main/java/org/openhab/automation/jrule/things/JRuleBridgeThing.java (hunk 1 above)
Comment (seime) [reply src/main/java/org/openhab/automation/jrule/things/JRuleBridgeThing.java:19]: I'll update the docs. The idea is that a bridge thing can have methods to list and manipulate sub things

File: src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java (hunk 2 above)
Comment (seime) [reply src/main/java/org/openhab/automation/jrule/internal/handler/JRuleHandler.java]: renamed to jrule-generated.jar

File: src/main/java/org/openhab/automation/jrule/things/JRuleBridgeThing.java (hunk 1 above)
Comment (seime) [reply src/main/java/org/openhab/automation/jrule/things/JRuleBridgeThing.java]: Updated

**Comment by seime** [comment]
Includes some fixes to issues introduced by #52 (https://github.com/seaside1/jrule/pull/59/commits/6aa7827d90e85ef1944a5291b7d2a50091a5f7e2 + https://github.com/seaside1/jrule/pull/59/commits/247983f4c3dc96bc16ddbae98c50e458206c4492)

**Pull request merged**